controllers/workspace/provision/deployment.go

```diff
@@ -433,6 +481,45 @@ func getWorkspaceSubpathVolumeMount(workspaceId string) corev1.VolumeMount {
 	return workspaceVolumeMount
 }
 
+func getVolumeWithConfigMap(name string) corev1.Volume {
+	modeReadOnly := int32(0640)
+	workspaceVolumeMount := corev1.Volume{
+		Name: name,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: name,
+				},
+				DefaultMode: &modeReadOnly,
+			},
+		},
+	}
+	return workspaceVolumeMount
+}
+
+func getVolumeWithSecret(name string) corev1.Volume {
+	modeReadOnly := int32(0640)
+	workspaceVolumeMount := corev1.Volume{
+		Name: name,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  name,
+				DefaultMode: &modeReadOnly,
+			},
+		},
+	}
+	return workspaceVolumeMount
+}
+
+func getVolumeMounts(segment, name string) corev1.VolumeMount {
+	workspaceVolumeMount := corev1.VolumeMount{
+		Name:      name,
+		ReadOnly:  true,
+		MountPath: path.Join("/etc/", segment, name),
```

Comment:
What Che currently has:
- an ability to customize is it should be mounted as env var or file;
- an ability to customize where the file should be mounted;
- an ability to specify container name where it should be injected into;

Reply:
and yeah, sorry for missing it before, the items above I got from Che Documentation https://www.eclipse.org/che/docs/che-7/end-user-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container/

Reply:
@tinakurian thanks for catching that item I referenced to

> an ability to specify container name where it should be injected into;

is actually deprecated. And instead users are able to define
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: git-credentials-secret
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: workspace-secret
  annotations:
    che.eclipse.org/automount-workspace-secret: false
```
and then mount such disabled by default secrets into the particular component with [automountWorkspaceSecrets](https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-workspace/src/main/resources/schema/1.0.0/devfile.json#L277).

Let's move this feature to a dedicated issue where we agreed on the desired model and workflow for devfile v2 and devworkspaces.

Reply:
I've created  Add an ability to select secret/config maps to mount for particular devworkspace/component https://github.com/devfile/devworkspace-operator/issues/435, so resolving this conversation.